src/polymer/expression-scanner.ts

```diff
@@ -201,6 +170,85 @@ export class DatabindingExpression {
   }
 }
 
+export class AttributeDatabindingExpression extends DatabindingExpression {
+  /**
+   * The element whose attribute/property is assigned to.
+   */
+  readonly astNode: parse5.ASTNode;
+
+  readonly databindingInto = 'attribute';
+
+  /**
+   * If true, this is databinding into the complete attribute. Polymer treats
+   * such databindings specially, e.g. they're setting the property by default,
+   * not the attribute.
+   *
+   * e.g.
+   * foo="{{bar}}" is complete, foo="hello {{bar}} world" is not complete.
+   *
+   * An attribute may have multiple incomplete bindings. They will be separate
+   * AttributeDatabindingExpressions.
+   */
+  readonly isCompleteBinding: boolean;
+
+  /** The databinding syntax used. */
+  readonly direction: '{'|'[';
+
+  /**
+   * If this is a two-way data binding, and an event name was specified
+   * (using ::eventName syntax), this is that event name.
+   */
+  readonly eventName: string|undefined;
+
+  /** The attribute we're databinding into. */
+  readonly attribute: parse5.ASTAttribute;
+
+  constructor(
+      astNode: parse5.ASTNode, isCompleteBinding: boolean, direction: '{'|'[',
+      eventName: string|undefined, attribute: parse5.ASTAttribute,
+      sourceRange: SourceRange, expressionText: string, ast: estree.Program) {
```

Comment:
That's some serious arity!  Not recommending a change per se, especially since this constructor is not really used by the outside world.  Definitely would be too much without type-checking! :)